hls-test-utils/src/Test/Hls.hs

```diff
@@ -447,6 +452,32 @@ runSessionWithServerInTmpDir' plugins conf sessConf caps tree act = withLock loc
     _fs <- FS.materialiseVFT tmpDir tree
     runSessionWithServer' plugins conf sessConf caps tmpDir act
 
+-- | Setup the test environment for isolated tests.
+--
+-- This creates a directory in the temporary directory that will be
+-- reused for running isolated tests.
+-- It returns the root to the testing directory that tests should use.
+-- This directory is not fully cleaned between reruns.
+-- However, it is totally safe to delete the directory between runs.
+--
+-- Additionally, this overwrites the 'XDG_CACHE_HOME' variable to isolate
+-- the tests from existing caches. 'hie-bios' and 'ghcide' honour the
+-- 'XDG_CACHE_HOME' environment variable and generate their caches there.
+setupTestEnvironment :: IO FilePath
+setupTestEnvironment = do
+  tmpDirRoot <- getTemporaryDirectory
+  let testRoot = tmpDirRoot </> "hls-test-root"
+      testCacheDir = testRoot </> ".cache"
+  createDirectoryIfMissing testRoot
+  createDirectoryIfMissing testCacheDir
+  setEnv "XDG_CACHE_HOME" testCacheDir
+  pure testRoot
+  where
+    createDirectoryIfMissing fp = do
```

Comment:
Why not use [createDirectoryIfMissing](https://hackage.haskell.org/package/directory-1.3.8.3/docs/System-Directory.html#v:createDirectoryIfMissing) from directory package? It can also create intermediate dirs on path, so can be called just once..

Reply:
That's a good suggestion :D My brain didn't want to see that function